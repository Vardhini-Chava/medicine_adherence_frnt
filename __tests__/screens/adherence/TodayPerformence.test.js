import Enzyme, { shallow,render } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { Provider } from "react-redux";
import React from "react";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import TodayPerformance from '../../../src/screens/adherence/TodayPerformance';

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore([]);
jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({ goBack: jest.fn() }),
    useFocusEffect: jest.fn().mockImplementation((func) => func()),
    useRoute: () => ({
      params: {
       im:{}
      }
    }),
  }));
describe("test collector category", () => {
  let store;
  store = mockStore({
    field: {
      name: "hello",
      value: "San",
    },
  });
  it("test category", () => {
    const wrapper = render(
      <Provider store={store}>
        <TodayPerformance  />
      </Provider>
    );

    expect.assertions(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('test share button', () => {
    const mockFn = jest.fn();
   const wrapper = shallow(<TodayPerformance updatetimes={mockFn} />);
   wrapper.find("#share")
   });
});