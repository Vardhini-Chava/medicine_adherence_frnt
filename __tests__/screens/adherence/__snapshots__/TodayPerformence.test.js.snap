// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test collector category test category 1`] = `
<view
  style="background-color:white;height:100%"
>
  <view
    pointer-events="box-none"
    style="position:absolute;left:0;right:0;align-items:center;justify-content:center;top:0;opacity:0;transform:[object Object]"
    testid="toastAnimatedContainer"
  >
    <view
      focusable="true"
      style="flex-direction:row;height:60px;width:340px;border-radius:6px;shadow-offset:[object Object];shadow-opacity:0.1px;shadow-radius:6px;elevation:2px;background-color:#FFF;border-left-width:5px;border-left-color:#69C779;opacity:1"
      testid="toastTouchableContainer"
    >
      <view
        style="0:[object Object]"
        testid="toastContentContainer"
      />
    </view>
  </view>
  <view
    style="padding:15px;background-color:lightgrey"
  >
    <text
      style="font-weight:bold"
    >
      Timings
    </text>
  </view>
  <view>
    <rctscrollview
      data=""
      scrolleventthrottle="50"
      stickyheaderindices=""
      testid="today"
      viewabilityconfigcallbackpairs=""
    >
      <view />
    </rctscrollview>
  </view>
</view>
`;
